types/types.go

```diff
@@ -208,7 +211,7 @@ type BlobInfoCache interface {
 
 	// RecordKnownLocation records that a blob with the specified digest exists within the specified (transport, scope) scope,
 	// and can be reused given the opaque location data.
-	RecordKnownLocation(transport ImageTransport, scope BICTransportScope, digest digest.Digest, location BICLocationReference)
+	RecordKnownLocation(transport ImageTransport, scope BICTransportScope, digest digest.Digest, compressorName string, location BICLocationReference)
```

Comment:
I didn't have a way to record the additional info that didn't end up requiring adding either a new parameter here, or a new method (which would be another write to the database) to the interface.  If you know of a way to do that, I'd be thrilled to restructure this.

Reply:
No _good_ way I’m afraid. `type BlobInfoCacheV2` + a type check to call V1 or V2 + probably an internal helper to simplify callers , _and_ all of the code supporting both at the same time.

Reply:
> This is an API break, sadly third-party implementations of `BlobInfoCache` are possible. (The API impact is the awkward part of the problem.)

I really wish you were wrong about that, but https://github.com/containers/buildah/pull/2828 exists.

> No _good_ way I’m afraid. `type BlobInfoCacheV2` + a type check to call V1 or V2 + probably an internal helper to simplify callers , _and_ all of the code supporting both at the same time.

Is this something you'd like me to pursue here?

Reply:
Okay, I'm going to start roughing that out: a v2 interface that, if the cache we're given implements it, we'll use to record both v1 and v2 information using different files for each.  For reads, it will fall back to checking for v1 data if the v2 interface doesn't return any data.

Reply:
Okay, I think that's going to work without being _too_ awful to look at.  This current revision reworks how we store the compression information - it's now in a bucket in the same file that mirrors the structure of the one we use for storing timestamp information.  Logic for updating two different files, and keeping track of their names, and initializing a boltdb cache using two files, got unwieldy pretty fast.  There's still some cleanup work to do to fix and flesh out the unit tests, but the functional parts are pretty much where I want them to be.